lib/mdl/ruleset.rb

```diff
@@ -42,14 +51,19 @@ def initialize
     end
 
     def rule(id, description, &block)
-      @rules[id] = Rule.new(id, description, block)
+      @rules[id] =
+        Rule.new(id, description, :fallback_docs => @fallback_docs, &block)
     end
 
     def load(rules_file)
       instance_eval(File.read(rules_file), rules_file)
       @rules
     end
 
+    def docs(url = nil, &block)
+      @fallback_docs = block_given? ? block : ->(_, _) { url }
```

Comment:
same

Reply:
Also I feel like I'm missing something obvious here, but how does the fall-thru work for ruleset docs vs rule docs?

Reply:
If it's not clear then maybe we need to rework it — let me know if this doesn't make sense?

The `docs_link` method only returns documentation from within the `Rule` instance (from the `@generate_docs` lambda) — the `Ruleset` that creates a `Rule` can pass on the docs block that it was configured with (as a fallback), but the `Rule` can always use `Rule#docs` to override that. Does that answer your question?

As an aside, there's potentially a change we should make here, currently this is valid, but a little counter-intuitive:

```ruby
rule 'MY003', 'Documents must start with B' do
  tags :opinionated
  docs 'https://example.com/override-docs' # <-- this currently works but makes no sense
  docs 'https://example.com/a-second-url'
  check do |doc|
    [1] if doc.lines[0] != 'B'
  end
end
```